lib/proxy_api/tftp.rb

```diff
@@ -1,5 +1,7 @@
 module ProxyAPI
   class TFTP < ProxyAPI::Resource
+    include Foreman::Renderer::Scope::Macros::Base
```

Comment:
This is very unclean. The Rendering scopes should only be used for template rendering. Can you please move dns resolotion to a dedicated service?

Reply:
Yeah I could not figure it out, so like new module and including it in both places?

Reply:
I think a service class would be better then module. Something like 
```
class DnsResolver
  def self.dns_lookup(name_or_ip)
    resolver = Resolv::DNS.new
     Timeout.timeout(Setting[:dns_conflict_timeout]) do
      begin
        resolver.getname(name_or_ip)
      rescue Resolv::ResolvError
        resolver.getaddress(name_or_ip)
      end
    end
  rescue StandardError => e
    log_warn "Template helper dns_lookup failed: #{e}"
    raise e
  end
end
```

and then here

```
DnsResolver.dns_lookup(URI.parse(@url).host)
```

and in dns_lookup macro

```
DnsResolver.dns_lookup(name_or_ip)
```